### PR TITLE
Added documentation for the configuration discovery of `cargo install` to the man pages

### LINKED ISSUE #11660: `cargo install` does not respect `.cargo/config.toml`
### Problem

When I run `cargo install ripgrep` in a directory with a `.cargo/config.toml` overriding the install root (like below), it installs to `$HOME/.cargo/bin` instead.

The interesting thing is that `cargo uninstall` reads the configured root correctly, so if I immediately run `cargo uninstall ripgrep` in the same directory, it complains that `package ID specification `ripgrep` did not match any packages`.

Contents of `.cargo/config.toml`:
```
[install]
root = "./apps"
```

### Steps

1. Create a config file like the one above.
2. Make an `apps` directory
3. Run `cargo install ripgrep` (or anything)

### Possible Solution(s)

_No response_

### Notes

Also tried nightly just now with the same result.

Obviously I can just `cargo install --root apps ...`, but I was trying to experiment with patching dependencies and building with different compile options and it seems `cargo install` is completely ignoring the configuration file.

### Version

```text
cargo 1.67.0 (8ecd4f20a 2023-01-10)
release: 1.67.0
commit-hash: 8ecd4f20a9efb626975ac18a016d480dc7183d9b
commit-date: 2023-01-10
host: x86_64-pc-windows-msvc
libgit2: 1.5.0 (sys:0.16.0 vendored)
libcurl: 7.86.0-DEV (sys:0.4.59+curl-7.86.0 vendored ssl:Schannel)
os: Windows 10.0.22621 (Windows 10 Pro) [64-bit]
```

### PR DESCRIPTION
**Pull request by jofas**
Fixes #11660.

### CODE REVIEW

File: src/doc/man/cargo-install.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ not start publishing `Cargo.lock` files until version 1.37, which means
 packages published with prior versions will not have a `Cargo.lock` file
 available.
 
+This command operates on system or user level, not project level.
```
Comment (weihanglo) [inline src/doc/man/cargo-install.md:72]: Not really orthogonal to the issue, just feel like the description becomes way too long at this moment. Could we have h3 headers for this paragraph, and maybe one for `Cargo.lock paragraph? I am think about something like “Dealing with lockfile" and “Configuration discovery”.

**Review by weihanglo — COMMENTED** [review]
You've written something please to read! Thank you! However, this is probably a breaking change for someone depending on old `cargo uninstall` behaviour. Could we split this PR into two, so we can approve `cargo install` part and discuss the other half afterward?

BTW, I think we may want to first warn the user who depends on the behaviour of implicit config discovery in `cargo uninstall`. After a transition period, we can then turn the warn into an error.

**Comment by jofas** [comment]
> However, this is probably a breaking change for someone depending on old cargo uninstall behaviour. Could we split this PR into two, so we can approve cargo install part and discuss the other half afterward?

Of course. Never contributed to a project that can't make a breaking change, release a new version and call it a day :sweat_smile:.

I'll split the breaking changes and make a new PR. Maybe we could use that PR to also tackle the fact that `install` starts configuration discovery at `$CARGO_HOME/config.toml`, whereas I think it would be more natural to anchor configuration discovery at `$CARGO_HOME/config.toml`, not start it, like you already pointed out in the issue:

https://github.com/rust-lang/cargo/blob/efd37336e94e2f98260ad28d89e509e02fe4a556/src/bin/cargo/commands/install.rs#L93-L94

---

> BTW, I think we may want to first warn the user who depends on the behaviour of implicit config discovery in cargo uninstall. After a transition period, we can then turn the warning into an error.

Can you point me to a location in the source where cargo already has such a warning? Then I could create a PR for the warning as well (of course only if that is something you'd want).

**Comment by weihanglo** [comment]
> Can you point me to a location in the source where cargo already has such a warning? Then I could create a PR for the warning as well (of course only if that is something you'd want).

You'll find them by searching “hard error” or “transition period” in this repository, such as:

https://github.com/rust-lang/cargo/blob/65cab34dc75587eeaff68d3c19358c4d79041452/src/cargo/ops/cargo_package.rs#L299-L306

**Review by weihanglo — APPROVED** [review]
Looks pretty good! Thanks for the update :)

**Comment by weihanglo** [comment]
BTW, @jofas could you squash these commits so that we won't have things added and removed making the history confusing? No problem if you're unable to do that.

**Comment by jofas** [comment]
Sure, I can squash the commits into a single one. Can you stop bors from merging?

**Comment by weihanglo** [comment]
BTW, if the bot detects more commits being pushed, it will cancel the merge automatically.

**Comment by jofas** [comment]
Thanks, good to know. I squashed the commits.

**Comment by weihanglo** [comment]
Thanks a lot!

@bors r+

**Pull request merged**